src/lambda/commands/deploySamApplication.ts

```diff
@@ -110,10 +105,13 @@ export async function deploySamApplication(
             ),
             deployApplicationPromise
         )
+
+        await deployApplicationPromise
```

Comment:
Are you sure we want/need synchronous behavior here? Could put `tryRemoveFolder()`  in `.finally()` .